examples/se_2_3_localization.cpp

```diff
@@ -0,0 +1,405 @@
+ *
+ * The system propagation noise covariance matrix becomes,
+ *    U = diagonal(0, 0, 0, sigma_omegax^2, sigma_omegay^2, sigma_omegaz^2, sigma_accx^2, sigma_accy^2, sigma_accz^2).
+ *
+ *  At the arrival of a exogeneous input u, the robot pose is updated
+ *  with X <-- X * Exp(u) = X + u.
+ *
+ *  Landmark measurements are of the range and bearing type,
+ *  though they are put in Cartesian form for simplicity.
+ *  Their noise n is zero mean Gaussian, and is specified
+ *  with a covariances matrix R.
+ *  We notice that the SE_2(3) action is the same as a
+ *  rigid motion action of SE(3).
+ *  This is the action of X \in SE_2(3) on a 3-d point b \in R^3 defined as,
+ *  X b = R b + p
+ *
+ *  Thus, the landmark measurements can be expressed as,
+ *  y = h(X,b) = X^-1 * b'
+ *
+ *      y_k = (brx_k, bry_k, brz_k)    // lmk coordinates in robot frame
+ *
+ *  We consider the beacons b_k situated at known positions.
+ *  We define the extended pose to estimate as X in SE_2(3).
+ *  The estimation error dx and its covariance P are expressed
+ *  in the tangent space at X.
+ *
+ *  All these variables are summarized again as follows
+ *
+ *    X   : robot's extended pose, SE_2(3)
+ *    u   : robot control input, u = u(X, y_imu) \in se_2(3) with X as state and y_imu = [alpha, omega] as IMU readings, see Eq. (3)
+ *    U   : control perturbation covariance
+ *    b_k : k-th landmark position, R^3
+ *    y   : Cartesian landmark measurement in robot frame, R^3
+ *    R   : covariance of the measurement noise
+ *
+ *  The motion and measurement models are
+ *
+ *    X_(t+1) = f(X_t, u) = X_t * Exp ( u )     // motion equation
+ *    y_k     = h(X, b_k) = X^-1 * b_k          // measurement equation
+ *
+ *  The algorithm below comprises first a simulator to
+ *  produce measurements, then uses these measurements
+ *  to estimate the state, using a Lie-based error-state Kalman filter.
+ *
+ *  This file has plain code with only one main() function.
+ *  There are no function calls other than those involving `manif`.
+ *
+ *  Printing simulated state and estimated state together
+ *  with an unfiltered state (i.e. without Kalman corrections)
+ *  allows for evaluating the quality of the estimates.
+ *
+ * A side note: Besides the approach described here in this illustration example,
+ * there are other interesting works like the paper,
+ * The Invariant Extended Kalman filter as a stable observer (https://arxiv.org/pdf/1410.1465.pdf)
+ * which assume a specific structure for the system propagation dynamics "f(X_t, u)" (group affine dynamics)
+ * that simplifies the covariance propagation and enables error dynamics with stronger convergence properties.
+ *
+ */
+
+#include "manif/SE_2_3.h"
+
+#include <Eigen/Dense>
+
+#include <vector>
+
+#include <iostream>
+#include <iomanip>
+
+using std::cout;
+using std::endl;
+
+using namespace Eigen;
+
+typedef Array<double, 3, 1> Array3d;
+typedef Matrix<double, 5, 1> Vector5d;
```

Comment:
This V5 is probably not needed since measurements and beacons are in R3

Reply:
Sorry I missed this.